pkg/controller/operands/kubevirt_test.go

```diff
@@ -247,6 +257,205 @@ var _ = Describe("KubeVirt Operand", func() {
 			Expect(foundResource.Data).To(Equal(outdatedResource.Data))
 			Expect(foundResource.Data).To(Not(Equal(expectedResource.Data)))
 		})
+
+		Context("Feature Gates", func() {
+			origKvFeatureGates := managedKvFeatureGates
+
+			const (
+				constFeatureGates = "DataVolumes,SRIOV,LiveMigration,CPUManager,CPUNodeDiscovery,Sidecar,Snapshot"
+				userModifiedFgs   = "userModifiedFg1,userModifiedFg2,userModifiedFg3"
+			)
+			BeforeEach(func() {
+				// list of fake KV managed FGs
+				managedKvFeatureGates = []string{fgEnabled, fgMissing, fgDisabled, fgNoChange}
+			})
+
+			AfterEach(func() {
+				managedKvFeatureGates = origKvFeatureGates
+			})
+
+			It("should not add the feature gates if they are not set in HyperConverged CR", func() {
+				existingResource := NewKubeVirtConfigForCR(hco, commonTestUtils.Namespace)
+				By("KV CR should contain the fgEnabled feature gate", func() {
+					Expect(existingResource.Data[virtconfig.FeatureGatesKey]).Should(Equal(constFeatureGates))
+				})
+			})
+
+			It("should add the feature gates if they are set in HyperConverged CR", func() {
+				hco.Spec.FeatureGates = map[string]bool{
+					fgEnabled:  true,
+					fgDisabled: false,
+				}
+
+				existingResource := NewKubeVirtConfigForCR(hco, commonTestUtils.Namespace)
+				By("KV CR should contain the fgEnabled feature gate", func() {
+					Expect(existingResource.Data[virtconfig.FeatureGatesKey]).Should(Equal(constFeatureGates + "," + fgEnabled))
+				})
+			})
+
+			It("should add feature gates if they are set to true", func() {
+				existingResource := NewKubeVirtConfigForCR(hco, commonTestUtils.Namespace)
+				By("Make sure the enabled FG is not there", func() {
+					Expect(existingResource.Data[virtconfig.FeatureGatesKey]).Should(Equal(constFeatureGates))
+				})
+
+				hco.Spec.FeatureGates = map[string]bool{
+					fgEnabled:  true,
+					fgDisabled: false,
+				}
+
+				cl := commonTestUtils.InitClient([]runtime.Object{hco, existingResource})
+				handler := (*genericOperand)(newKvConfigHandler(cl, commonTestUtils.GetScheme()))
+				res := handler.ensure(req)
+				Expect(res.UpgradeDone).To(BeFalse())
+				Expect(res.Updated).To(BeTrue())
+				Expect(res.Overwritten).To(BeFalse())
+				Expect(res.Err).To(BeNil())
+
+				foundResource := &corev1.ConfigMap{}
+				Expect(
+					cl.Get(context.TODO(),
+						types.NamespacedName{Name: existingResource.Name, Namespace: existingResource.Namespace},
+						foundResource),
+				).To(BeNil())
```

Comment:
This code block has been reused in the next 3 tests, ca we move it to a separate helper function under the whole context?